include/ddc/kernels/splines/spline_traits.hpp

```diff
@@ -0,0 +1,119 @@
+
+template <typename T>
+inline constexpr bool is_spline_builder_v = is_spline_builder<T>::value;
+
+template <class T>
+struct is_spline_builder2D : std::false_type {};
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSpline1,
+        class BSpline2,
+        class DDimI1,
+        class DDimI2,
+        ddc::BoundCond BcLower1,
+        ddc::BoundCond BcUpper1,
+        ddc::BoundCond BcLower2,
+        ddc::BoundCond BcUpper2,
+        ddc::SplineSolver Solver>
+struct is_spline_builder2D<SplineBuilder2D<ExecSpace, MemorySpace, BSpline1, BSpline2, DDimI1, DDimI2, BcLower1, BcUpper1, BcLower2, BcUpper2, Solver>> : std::true_type {};
+
+template <typename T>
+inline constexpr bool is_spline_builder2D_v = is_spline_builder2D<T>::value;
+
+template <class T>
+struct is_spline_evaluator : std::false_type {};
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSplines,
+        class EvaluationDDim,
+        class LowerExtrapolationRule,
+        class UpperExtrapolationRule>
+struct is_spline_evaluator<SplineEvaluator<ExecSpace, MemorySpace, BSplines, EvaluationDDim, LowerExtrapolationRule, UpperExtrapolationRule>> : std::true_type {};
+
+template <typename T>
+inline constexpr bool is_spline_evaluator_v = is_spline_evaluator<T>::value;
+
+template <class T>
+struct is_spline_evaluator2D : std::false_type {};
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSpline1,
+        class BSpline2,
+        class EvaluationDDim1,
+        class EvaluationDDim2,
+        class LowerExtrapolationRule1,
+        class UpperExtrapolationRule1,
+        class LowerExtrapolationRule2,
+        class UpperExtrapolationRule2>
+struct is_spline_evaluator2D<SplineEvaluator2D<ExecSpace, MemorySpace, BSpline1, BSpline2, EvaluationDDim1, EvaluationDDim2, LowerExtrapolationRule1, UpperExtrapolationRule1, LowerExtrapolationRule2, UpperExtrapolationRule2>> : std::true_type {};
+
+template <typename T>
+inline constexpr bool is_spline_evaluator2D_v = is_spline_evaluator2D<T>::value;
+
+template <class Builder, class Evaluator, class Enable = void>
+struct is_spline_compatible : std::false_type {};
+
+template <class Builder, class Evaluator>
+struct is_spline_compatible<Builder, Evaluator,
+       std::enable_if_t<is_spline_builder_v<Builder> && is_spline_evaluator_v<Evaluator>
+                        || is_spline_builder_v<Evaluator> && is_spline_evaluator_v<Builder> >> {
+       static constexpr bool value = std::is_same_v<typename Builder::exec_space, typename Evaluator::exec_space> && 
+                                     std::is_same_v<typename Builder::memory_space, typename Evaluator::memory_space> &&
+                                     std::is_same_v<typename Builder::bsplines_type, typename Evaluator::bsplines_type>;
+};
+
+template <class Builder, class Evaluator>
+struct is_spline_compatible<Builder, Evaluator,
```

Comment:
I'm not sure this name is very explicit. I would expect it to check if a spline is compatible with a builder or if a spline is compatible with an evaluator, not if a spline builder is compatible with an evaluator

I'm not sure what to suggest instead though
```suggestion
struct is_spline_interpolation_possible<Builder, Evaluator,
```
```suggestion
struct is_interpolation_possible<Builder, Evaluator,
```
?

Reply:
I do not have a strong preference.
But, I would prefer to include a term `spline` at least and ideally not too long

Reply:
I definitely agree that it should not be too long.

I'm not sure it is important to include `spline` though? If you had a different intermediate representation it is likely that you would still need a similar check for a compatible Builder and Evaluator. That is why I suggested `is_interpolation_possible`

If splines are important then maybe `shares_spline_basis`? But this doesn't cover the Memory/Execution space aspects